server/setup_flow.go

```diff
@@ -633,9 +633,20 @@ func (p *Plugin) submitCreateCloudOAuthInstance(f *flow.Flow, submission map[str
 		return "", nil, nil, errors.New("no Jira OAuth Client Secret is present in the request")
 	}
 
-	jiraURL, instance, err := p.installCloudOAuthInstance(jiraURL, clientID, clientSecret)
+	existingInstance, err := p.instanceStore.LoadInstance(types.ID(jiraURL))
 	if err != nil {
-		return "", nil, nil, err
+		return "", nil, nil, errors.Wrap(err, "failed to load existing cloud-oauth instance")
+	}
+
+	instance, ok := existingInstance.(*cloudOAuthInstance)
+	if !ok {
+		return "", nil, nil, errors.Errorf("existing instance is not a cloud-oauth instance. ID: %s", jiraURL)
```

Comment:
Why does the `submitCreateCloudOAuthInstance` assume that there's already an existing instance of cloud-oauth? Isn't the `submitCreateCloudOAuthInstance` what get's triggered once the admin has clicked the submit button? That should work regardless of whether there is an existing instance.